pkg/controller/monitor/monitor_controller.go

```diff
@@ -265,6 +294,64 @@ func (r *ReconcileMonitor) Reconcile(ctx context.Context, request reconcile.Requ
 		return reconcile.Result{}, err
 	}
 
+	managementClusterConnection, err := utils.GetManagementClusterConnection(ctx, r.client)
+	if err != nil {
+		log.Error(err, "Error reading ManagementClusterConnection")
+		r.status.SetDegraded("Error reading ManagementClusterConnection", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	// In managed clusters with certificate management disabled, successful reconciliation of non-NetworkPolicy resources
+	// in the monitor controller ensures that NetworkPolicy is reconcilable (by enabling* the creation of containing Tier
+	// and License). Therefore, to prevent a chicken-and-egg scenario, we only reconcile non-NetworkPolicy resources once
+	// we can confirm that all requirements to reconcile NetworkPolicy have been met.
```

Comment:
>  we only reconcile non-NetworkPolicy resources once we can confirm that all requirements to reconcile NetworkPolicy have been met.

Do you mean we only reconcile NetworkPolicy once all the requirements have been met? We should reconcile non-NP resources unconditionally.

If so, isn't it enough to simply put the NetworkPolicy component at the end, conditional on whether or not the v3 NetworkPolicy API and allow-tigera tier is present in the cluster?

Reply:
Yep, I did mean NetworkPolicy, thanks for catching that. As discussed, we'll move towards an approach where controllers that impact NetworkPolicy reconciliation reconcile NetworkPolicy last, and check whether or not they can reconcile based on tier availability (without the use of a helper function).

Regarding the comment typo: Will resolve in current PR.

Regarding the controller update: Flagged for resolution in the immediate follow-up PR.